File: app/api/internal/tree/tree.go
package tree

import "context"

type Entry struct {
	Path string `json:"path"`
	Name string `json:"name"`
	Type string `json:"type"`
	Size int64  `json:"size"`
	SHA  string `json:"sha"`
}

type Lister interface {
	ListTree(ctx context.Context, owner, name, path string) ([]Entry, error)
}

File: app/web/src/api/tree.ts
import type { TreeListResponse } from "./types";

export async function fetchTree(owner: string, name: string, path = "") {
  const url = `/api/v1/repos/${owner}/${name}/tree?path=${encodeURIComponent(path)}`;
  const res = await fetch(url);

  if (!res.ok) {
    throw new Error(`Request failed with status ${res.status}`);
  }

  const json = (await res.json()) as TreeListResponse;
  return json.data;
}
